lib/platform/common.ts

```diff
@@ -142,6 +142,8 @@ export interface Platform {
   findIssue(title: string): Promise<Issue | null>;
   getIssueList(): Promise<Issue[]>;
   getVulnerabilityAlerts(): Promise<VulnerabilityAlert[]>;
+  getJsonFile(fileName: string): Promise<any | null>;
+  initRepoConfig(config: RepoParams): Promise<RepoResult>;
```

Comment:
I hadn't planned for this. Why is it needed?

Reply:
The latest step of `initRepo` is repo cloning, if all pre-conditions are met. If we check `optimizeForDisabled` after this `initRepo`, it would be too late. So I thought if we want to extract platform-independent checks, we have to split the functions into (1) HTTP-related things that initalize local config data with `initRepoConfig()`, (2) check platform-independent conditions and (3) do more side-effecting things like clone/fork with `initRepo()`.

Reply:
Also I'm just not sure of what the consequences can be if we make these checks right after `initRepo()`

Reply:
No, cloning is lazy now so calling `git.initRepo()` does not cause an immediately clone

Reply:
Oh, I get it 🤦

Reply:
When I re-introduce init/config caching, it will mean that this extra getJsonFile() can be skipped too if the default branch commit is unchanged (which happens for most forks, which never get updated). The flow would be:

1. initRepo to platform, in the result we learn the default branch name
2. initRepo to git does not clone but just fetches the list of branches and SHAs
3. We then get the default branch's SHA and compare against our cached init cache
4. If the default SHA is unchanged, we reuse the init cache and don't need to clone or fetch the renovate.json
5. If the SHA is changed, we fetch the renovate.json (if fork or optimizing for disabled), then clone the repo if continuing, etc

So the key part of this PR is to remove the fetchJsonFile call from step (1) so that it can be done in step (5). It's ok to keep the ordering as-is for now, because I haven't done the config caching yet